math/perfectnumber_test.go

```diff
@@ -0,0 +1,157 @@
+package math_test
+
+import (
+	"testing"
+
+	"github.com/TheAlgorithms/Go/math"
+)
+
+type sumOfProperDivisorsExample struct {
```

Comment:
```suggestion
type testCase struct {
```

Reply:
I am not sure if this is a good idea to use a _generic_ name like `testCase` here, since there are more functions returning some test cases in this file. Maybe getting rid of the type `sumOfProperDivisorsExample` and returning a `map[uint]uint` from the function `getSumOfProperDivisorsExamples()` would be an option? Something like:

```golang
func getSumOfProperDivisorsExamples() map[uint]uint {
	return map[uint]uint{
		0:     0,
		1:     0,
		2:     1,
		3:     1,
               //...
	}
}

func TestSumOfProperDivisors(t *testing.T) {
	for input, expected := range getSumOfProperDivisorsExamples() {
		actual := math.SumOfProperDivisors(input)
		if actual != expected {
			t.Errorf(
				"Expected SumOfProperDivisors(%d) to be: %d, but got: %d",
				input, expected, actual)
		}
	}
}
```

Reply:
My point here was not to have generic name `testCase` but rather a prefix - I used that same comment in many places and in many of your PR's since you tend to use the `Example` suffix a lot. The suffix you use is `Example` which is not really an example but a case for a test - it should also contain bad cases to make sure that they test the bad route of a function.

Use your own judgement here, if there are two things that could be called a testCase then add more context to it. For example, `testCaseSumOfProperDivisors` though I dislike the long name, it's an option.

>  Maybe getting rid of the type sumOfProperDivisorsExample and returning a map[uint]uit from the function getSumOfProperDivisorsExamples() would be an option?

Do you really think it helps someone reading your code?

I would also recommend adding more context in the other PR's where you've added `testCase` to something like `testCaseXyz`.